src/Swagger/Serializer/DocumentationNormalizer.php

```diff
@@ -302,12 +303,22 @@ private function updateGetOperation(\ArrayObject $pathOperation, array $mimeType
             return $pathOperation;
         }
 
+        $pathIdentifier = 'id';
+        $pathType = 'string';
+        foreach ($this->propertyNameCollectionFactory->create($resourceClass, []) as $propertyName) {
+            $property = $this->propertyMetadataFactory->create($resourceClass, $propertyName);
+            if ($property->isIdentifier()) {
+                $pathIdentifier = $propertyName;
+                $pathType = $property->getType()->getBuiltinType();
+            }
+        }
```

Comment:
We have an issue on composite identifiers here. Indeed, a composite identifier would be written as a string (for example `foo=1;bar=2`). With this code, the identifier would only be `bar`./
About the `id` name, I think it's like this so that it's being used in the path from Swagger. I'm not sure how Swagger is identifying the url parameter like this.

You could maybe reuse the same code from a new method instead of repeating it.